### PR TITLE
Apply interface and factory pattern to `HackAndSlash` action 

### PR DESCRIPTION
**Pull request by U-lis**
This change affects to `NineChronicles` and `NineChronicles.Headless`.

For convenience, I just applied to latest two version of HAS actions.
Should I apply to all versions of HAS action?

### CODE REVIEW

File: Lib9c/Action/Interface/IHackAndSlash.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using Libplanet.Action;
+
+namespace Nekoyume.Action.Interface
+{
+    public interface IHackAndSlash: IAction
+    {
+
+    }
```
Comment (moreal) [inline Lib9c/Action/Interface/IHackAndSlash.cs:8]: This is not important, only for styling.
```suggestion
    public interface IHackAndSlash : IAction
    {
    }
```

File: Lib9c/Action/Factory/HackAndSlashFactory.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        public static IHackAndSlash CreateByBlockIndex(
+            long blockIndex,
+            Address avatarAddress,
+            int worldId,
+            int stageId,
+            List<Guid> costumeIds = null,
+            List<Guid> equipmentIds = null,
+            List<Guid> consumableIds = null,
+            List<RuneSlotInfo> runeSlotInfos = null,
```
Comment (moreal) [inline Lib9c/Action/Factory/HackAndSlashFactory.cs:18]: How about enabling the `#nullable enable` feature and turning those types as `List<T>?`?

**Review by hskim881028 — APPROVED** [review]

File: Lib9c/Action/Factory/HackAndSlashFactory.cs (hunk 2 above)
Comment (U-lis) [reply Lib9c/Action/Factory/HackAndSlashFactory.cs:18]: > How about enabling the `#nullable enable` feature and turning those types as `List<T>?`?

Is it OK to set those values as just `List<T>`, not `List<Guid>` or `List<RuneSlotInfo>`?
I don't have deep knowledge about this, so I need your help :pray:

Comment (moreal) [reply Lib9c/Action/Factory/HackAndSlashFactory.cs:18]: Ah, I mean `List<Guid>` and `List<RuneSlotInfo>` but with the `?` suffix.

The nullable types like `List<Guid>?` and `List<RuneSlotInfo>?`.

**Comment by U-lis** [comment]
Merging `development` branch makes large change history to this branch :thinking: 
I don't know why development-based commits are shown here..
Please check only following files:
- [Lib9c/Action/Interface/IHackAndSlash.cs](https://github.com/planetarium/lib9c/pull/1590/files#diff-060723af165d4a9082c1ba3d87d64d0e511f6f753068025ca9873162f0e89280)
- [Lib9c/Action/Factory/HackAndSlashFactory.cs](https://github.com/planetarium/lib9c/pull/1590/files#diff-e339099269f05be9027056b76c243df2f3cbc235d75b852c58078f5cc16973f0)
- [.Lib9c.Tests/Action/Factory/HachAndSlashFactoryTest.cs](https://github.com/planetarium/lib9c/pull/1590/files#diff-3b00a2911320b5f2fe51893a75e06db65516b473a08d93013abae10d49b3d0c8)

**Review by boscohyun — APPROVED** [review]

**Pull request merged**